src/api.py

```diff
@@ -64,73 +64,66 @@ def storage_urls():
         except Exception as e:
             return Response(status_code=400, error_message=e).json()
 
-    @app.route("/blob", methods=["GET", "POST"])
-    def blob():
-
-        if request.method == "GET":
-            try:
-                user_search = request.args.to_dict()
-                if not "blob_type" in user_search.keys():
-                    return Response(
-                        status_code=400, error_message="Must provide blob_type"
-                    ).json()
-                if not f.validate_search_fields(user_search):
-                    return Response(
-                        status_code=400,
-                        error_message="KeyError: invalid blob_type or search field",
-                    ).json()
-                # blob_type only - return all entries for blob_type
-                with Session() as session:
-                    if len(user_search) == 1:
-                        matches = f.all_entries(user_search["blob_type"], session)
-                        return Response(body=matches).json()
-                    else:
-                        matches: dict = f.search_metadata(
-                            user_search["blob_type"], user_search, session
-                        )
-                    return Response(body=matches).json()
-            except Exception as e:
+    @app.get("/blob")
+    def read_blob(request: Request):
+        try:
+            # user_search = request.query_params
+            user_search = request.query_params
+            if not "blob_type" in user_search.keys():
```

Comment:
can you just use a pydantic class to do this confirmation, and return whatever error code it provides?

Reply:
i'm not sure I can because UserSearch is a special kind of class already: class 'starlette.datastructures.QueryParams'